module_utils/ansible_nailgun_cement.py

```diff
@@ -40,6 +40,30 @@ def create_server(server_url, auth, verify_ssl):
     ).save()
 
 
+# Prerequisite: create_server
+def ping_server(module):
+    try:
+        return entities.Ping().search_json()
+    except Exception as e:
+        module.fail_json(msg="Failed to connect to Foreman server: %s " % e)
+
+
+def update_fields(new, old, fields):
+    needs_update = False
+    for field in fields:
+        if hasattr(new, field) and hasattr(old, field):
+            new_attr = getattr(new, field)
+            old_attr = getattr(old, field)
+            id_attrs = hasattr(new_attr, 'id') and hasattr(old_attr, 'id')
+            if old_attr is None or (id_attrs and new_attr.id != old_attr.id) or (not id_attrs and new_attr != old_attr):
+                setattr(old, field, new_attr)
+                needs_update = True
+        elif hasattr(old, field) and getattr(old, field) is not None and not hasattr(new, field):
+            setattr(old, field, None)
+            needs_update = True
+    return needs_update, old
+
```

Comment:
This does not actually call update() on the entity, it just updates the fields in memory. That's kind of a large difference.